common/api_adapter.js

```diff
@@ -32,10 +33,32 @@ async function getHeaders(url, authorization) {
 async function getJsonFrom(url, authorization, retryAttempt = 1, delay = INITIAL_BACKOFF_DELAY) {
     try {
         const {body} = await get('get', url, authorization);
-        return body;
+
+        /*
+         handle the case when Martha receives empty JSON body. The reason behind forming a response with status 500
+         and throwing it in `try` is so that it can be caught locally and be retried.
+         */
+        if (Object.keys(body).length === 0) {
+            console.log(`Received an empty JSON body while trying to resolve url '${url}'. Creating a response with ` +
+            'status 500 and will retry.');
```

Comment:
Is a retry guaranteed? I assume there are not infinite retries, and will at some point stop.

Alternatively, we could just log the `retryAttempt`, say:

```
`status 500 during attempt ${retryAttempt}`
```

Reply:
>Is a retry guaranteed? I assume there are not infinite retries, and will at some point stop.

yes. It will be caught locally, and that [catch block](https://github.com/broadinstitute/martha/blob/942005e4a869c34cdb51f3ba626c929290ddbf6e/common/api_adapter.js#L41-L55) has retry logic which also checks the retry attempt. So it should not be retried infinite times. The idea is that if receiving an empty json body was just once and not consistent, retrying would help in this case. We can certainly add the retry attempt in the log message.
>Alternatively, we could just log the retryAttempt

We can also add that. I will add it in the log message

Reply:
I misunderstood the comment 🤦‍♀. Yes it does make more sense to log the retry attempt instead of saying `will retry` every time as that would not be true on the last attempt.